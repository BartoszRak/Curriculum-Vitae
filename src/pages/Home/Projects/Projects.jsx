import React, { useMemo } from 'react'
import PropTypes from 'prop-types'
import { withStyles, Typography } from '@material-ui/core'
import { injectIntl, FormattedMessage } from 'react-intl'

import ProjectsList from './ProjectsList'
import { useFavoriteProjectsStorage } from '~hooks'

import styles from './Projects.style'
import technologiesMap from './TechnologiesMap.json'

export function Projects({ classes, intl }) {
  const { formatMessage } = intl
  const { getProjects } = useFavoriteProjectsStorage()
  const favoriteProjects = getProjects() || []

  const projects = useMemo(() => {
    return [
      {
        description: formatMessage({
          id: 'pages.home.projects.projects.insectifyApi.description',
        }),
        tags: [
          technologiesMap.nestjs,
          technologiesMap.express,
          technologiesMap.firebase,
        ],
        title: formatMessage({
          id: 'pages.home.projects.projects.insectifyApi.title',
        }),
        url: 'https://github.com/BartoszRak/Insectify-api',
      },
      {
        description: formatMessage({
          id: 'pages.home.projects.projects.reactAwesomeBoilerplate.description',
        }),
        tags: [
          technologiesMap.react,
          technologiesMap.redux,
          technologiesMap.rematch,
        ],
        title: formatMessage({
          id: 'pages.home.projects.projects.reactAwesomeBoilerplate.title',
        }),
        url: 'https://github.com/BartoszRak/React-Awesome-Boilerplate',
      },
      {
        description: formatMessage({
          id: 'pages.home.projects.projects.mx.description',
        }),
        tags: [technologiesMap.typescript],
        title: formatMessage({ id: 'pages.home.projects.projects.mx.title' }),
        url: 'https://github.com/BartoszRak/Mx',
      },
    ]
      .map(rawProject => ({
        ...rawProject,
        isFavorite: Boolean(
          favoriteProjects.findIndex(project => project === rawProject.url) !== -1
        ),
      }))
      .sort(a => {
        if (a.isFavorite) return -1
        return 1
      })
  }, [favoriteProjects])

  return (
    <div className={classes.root} id="projects">
      <Typography className={classes.title} color="secondary" variant="h1">
        <FormattedMessage id="pages.home.projects.title" />
        <span aria-hidden className={classes.emoji}>
          {' '}
          ðŸ˜Ž
        </span>
      </Typography>
      <ProjectsList data={projects} />
    </div>
  )
}

Projects.propTypes = {
  classes: PropTypes.object,
  intl: PropTypes.object,
}

export default injectIntl(withStyles(styles)(Projects))
