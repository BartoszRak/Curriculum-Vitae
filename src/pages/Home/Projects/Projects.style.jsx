export default () => ({
  root: {
    width: '100%',
  },
  title: {
    fontSize: '3rem',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
  },
})
