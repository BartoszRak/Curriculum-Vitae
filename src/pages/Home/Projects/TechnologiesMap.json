{
  "react": {
    "name": "React",
    "url": "https://reactjs.org/"
  },
  "rematch": {
    "name": "Rematch",
    "url": "https://github.com/rematch/rematch"
  },
  "redux": {
    "name": "Redux",
    "url": "https://redux.js.org/"
  },
  "typescript": {
    "name": "TypeScript",
    "url": "https://www.typescriptlang.org/"
  },
  "express": {
    "name": "Express",
    "url": "https://expressjs.com//"
  },
  "nestjs": {
    "name": "Nestjs",
    "url": "https://nestjs.com/"
  },
  "firebase": {
    "name": "Firebase",
    "url": "https://firebase.google.com/"
  },
  "graphql": {
    "name": "GRaphQL",
    "url": "https://graphql.org/"
  }
}