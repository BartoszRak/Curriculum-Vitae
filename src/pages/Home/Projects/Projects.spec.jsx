import React from 'react'
import { shallow } from 'enzyme'

import { Projects } from './Projects'

jest.mock('./ProjectsList', () => 'ProjectsListMock')

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Projects {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Projects', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
