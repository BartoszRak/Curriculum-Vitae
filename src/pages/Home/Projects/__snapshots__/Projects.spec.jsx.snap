// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Projects should render as expected 1`] = `
<div
  id="projects"
>
  <TypographyMock
    color="secondary"
    variant="h1"
  >
    What have I created? ðŸ˜Ž
  </TypographyMock>
  <ProjectsListMock />
</div>
`;
